CHANGELOG/1.2-CHANGELOG.md

```diff
@@ -2,9 +2,619 @@
 
 ## Table of Contents
 
+- [v1.2.0](#v120)
 - [v1.2.0-rc2](#v120-rc2)
 - [v1.2.0-rc1](#v120-rc1)
 
+# v1.2.0
+
+## Breaking Changes
+
+### `GRPCRoute` and `ReferenceGrant` `v1alpha2` removal
+
+As per a previous deprecation notice, in this version, both Experimental
+and Standard channel CRDs will no longer serve the `v1alpha2` versions
+of `GRPCRoute` and `ReferenceGrant`.
+
+* Disable v1alpha2 versions for GRPCRoute and ReferenceGrant by @youngnick in
+  [#3361](https://github.com/kubernetes-sigs/gateway-api/pull/3361)
+
+#### Upgrades
+
+Before upgrading to Gateway API v1.2, you'll want to confirm that any
+implementations of Gateway API have been upgraded to support the `v1` API
+version of these resources instead of the `v1alpha2` API version. Note that
+even if you've been using `v1` in your YAML manifests, a controller may still be
+using `v1alpha2` which would cause it to fail during this upgrade.
+
+Once you've confirmed that the implementations you're relying on have upgraded
+to v1, it's time to install the v1.2 CRDs. In most cases, this will work without
+any additional effort.
+
+If you ran into issues installing these CRDs, it likely means that you have
+`v1alpha2` in the `storedVersions` of one or both of these CRDs. This field is
+used to indicate which API versions have ever been used to persist one of these
+resources. Unfortunately, this field is not automatically pruned. To check these
+values, you can run the following commands:
+
+```
+kubectl get crd grpcroutes.gateway.networking.k8s.io -ojsonpath="{.status.storedVersions}"
+kubectl get crd referencegrants.gateway.networking.k8s.io -ojsonpath="{.status.storedVersions}"
+```
+
+If either of these return a list that includes "v1alpha2", it means that we need
+to manually remove that version from `storedVersions`.
```

Comment:
@mlavacca I ran out of time to finish this part - do you mind adding some instructions here? I think the gist would be:

1. A link to https://github.com/kubernetes-sigs/kube-storage-version-migrator
2. Some kind of simple script that makes a no-op update to all GRPCRoutes and ReferenceGrants in the cluster to make sure they've all been migrated to the latest storage version (likely simpler than running storage-version-migrator)
3. A kubectl patch command that updates `status.storedVersions` to drop v1alpha2